pandas/core/groupby.py

```diff
@@ -4429,7 +4429,13 @@ def _groupby_indices(values):
         # bit better than factorizing again
         reverse = dict(enumerate(values.categories))
         codes = values.codes.astype('int64')
-        _, counts = _hash.value_count_int64(codes, False)
+        keys, counts = _hash.value_count_int64(codes, False)
+
+        if not counts.size == values.categories.size:
```

Comment:
The _algos.groupby_indices function expects a counts array that is the same length as the categories. If one of the categories has no values in it (empty bins), the _hash.value_count_int64 function would just return a shorter count array (with no 0 entries for empty categories/bins). My initial approach was to pad the array with 0s in the right places, but per chris-b1's suggestion, I used np.bincount instead.